### PR TITLE
Added Hungarian mode

### PR DESCRIPTION
**Pull request by krmarci**
I added a Hungarian mode based on the guide made by the Hungarian Tolkien Society.

P. S. When typing a capital Á, É, Í, Ó, Ú, Ő, Ű, the ára does not get capitalized (in any mode). I wonder whether that's intentional or a bug.

### CODE REVIEW

**Comment by krmarci** [comment]
There are two more things I'm wondering about.

1. Traditionally spelled names are quite frequent in Hungarian. I'm not sure how (or if) to indicate traditional spellings when transcribing, or to stick to a purely phonetic transcription.
2. The author of the original article makes the choice to use hwesta and unque for c and dz, the voiceless and voiced alveolar affricates. However, in the Tengwar chart, the alveolar sounds are all located in the first column. The use of thúle and anto for c and dz would make much more sense to me (given that these sounds do not exist in Hungarian), the only reason the author decides to put them in column 4 is that þ and đ are more common sounds in English. I'm not sure whether to deviate a little bit from the article here...

**Comment by arnog** [comment]
Thank you for your contribution!

I am not a Hungarian speaker, so please take my comments with a grain of salt.
Looking at [Omniglot](https://omniglot.com/conscripts/tengwar_hu.php#:~:text=The%20Tengwar%20mode%20for%20Hungarian,following%20the%20Tengwar%20phonetic%20rules.) I notice there are some differences. They may be justified, but I wanted to make you aware of them for your consideration. 

Using the Sample Text as a reference:
- The "ü" in "születik" is transcribed either with a `double-dot-above` or a `breve`
- "egyenlő" uses a nazalization mark or not
- "ő" in "egyenlő" uses a `double-dot-above` or a `breve`
- "ii" in "lelkiismerettel" uses two `dot-above` or  a `double-dot-above`
- "bírván" uses `roomen` or `ore`

**Comment by krmarci** [comment]
Thanks for your response! Luckily, the differences between the two versions seem to be minor.

ö/ő and ü/ű essentially use the unreserved tehtar in both variants. Füzessy (the one I found) and Radványi (the one you found) just used it the other way round.

Radványi does not give any explanation about the nasalization process, in contrast to Füzessy. Füzessy uses the tilde in cases where the nasal consonant is in the same column as the consonant after it.

The "ii" in lelkiismerettel seems to be an odd coincidence. It is a composite word of lelki and ismeret (+ the instrumental/comitative ending -vel), with the two i's occurring exactly at the boundary.
The same thing seems to happen elsewhere as well, when two identical vowels are next to each other: ee, eé, ii, ií, oo, oó, uu, uú, üü and üű. As long vowels in Hungarian are marked with diacritics, and the only place where identical vowels appear next to each other are word composition boundaries, it would make sense to me to eliminate these for Hungarian, and make the first one appear on a telco, and the second one on the following consonant. I'll fix that soon.

The r-rule does not make much sense for Hungarian. While in - for example - English and German, final r sounds differently to an intervocalic one, there is only one r sound in Hungarian.

Based on Radványi, my reasoning about the placement of c/dz was not that far-fetched after all. I'll stick to his one (thúle and anto for c/dz) for future versions.

To sum up, with the exception of c/dz, the Füzessy version seems to be more well thought out. I'll send a fix soon.

**Comment by arnog** [comment]
Excellent, thanks for your thoughtful answer. Like  I said, I'm not qualified to judge the accuracy, but just wanted to make sure you had considered this.
I'll merge the mode file now and will be adding it to Tecendil shortly.

**Comment by krmarci** [comment]
Wonderful, thank you very much! :-)

**Comment by arnog** [comment]
Tecendil has been updated with Hungarian mode support.

BTW, if you feel so inclined, it is possible to localize Tecendil in Hungarian by adding to the `https://github.com/arnog/tecendil-js/blob/master/strings.json` file.
